File: app/src/main/java/com/alexanderPodkopaev/dev/behancer/ui/profile/ProfileActivity.java
package com.alexanderPodkopaev.dev.behancer.ui.profile;

import android.support.v4.app.Fragment;

import com.alexanderPodkopaev.dev.behancer.common.SingleFragmentActivity;

/**
 * Created by Vladislav Falzan.
 */

public class ProfileActivity extends SingleFragmentActivity {

    public static final String USERNAME_KEY = "USERNAME_KEY";

    @Override
    protected Fragment getFragment() {
        if (getIntent() != null) {
            return ProfileFragment.newInstance(getIntent().getBundleExtra(USERNAME_KEY));
        }
        throw new IllegalStateException("getIntent cannot be null");
    }
}

File: app/src/main/java/com/alexanderPodkopaev/dev/behancer/ui/profile/ProfileActivity.kt
package com.alexanderPodkopaev.dev.behancer.ui.profile

import androidx.fragment.app.Fragment
import com.alexanderPodkopaev.dev.behancer.AppDelegate
import com.alexanderPodkopaev.dev.behancer.common.SingleFragmentActivity
import com.alexanderPodkopaev.dev.behancer.data.Storage


class ProfileActivity : SingleFragmentActivity() {
    protected override val fragment: Fragment
        get() {
            if (intent != null) {
                return ProfileFragment.newInstance(intent.getBundleExtra(USERNAME_KEY))
            }
            throw IllegalStateException("getIntent cannot be null")
        }


    companion object {
        const val USERNAME_KEY = "USERNAME_KEY"
    }
}